tests/NLog.UnitTests/Config/RuleConfigurationTests.cs

```diff
@@ -589,5 +591,221 @@ public void LoggingRule_LevelOff_NotSetAsActualLogLevel()
+                { "Debug", LogLevel.Debug },
+                { "Warn", LogLevel.Warn },
+                { "Error", LogLevel.Error },
+                { " error", LogLevel.Error },
+                { "Fatal", LogLevel.Fatal },
+                { "Wrong", LogLevel.Off },
+            };
+            foreach (var logLevel in logLevels)
+            {
+                yield return new object[] { logLevel.Key, new LogLevel[] { logLevel.Value } };
+
+                for (int bonusLevelIdx = LogLevel.MinLevel.Ordinal; bonusLevelIdx <= LogLevel.MaxLevel.Ordinal; ++bonusLevelIdx)
+                {
+                    var bonusLevel = LogLevel.FromOrdinal(bonusLevelIdx);
+                    yield return new object[] { logLevel.Key + ", " + bonusLevel.ToString(), new LogLevel[] { logLevel.Value, bonusLevel } };
+                }
+            }
+        }
+
+        [Theory]
+        [MemberData(nameof(LoggingRule_MinMaxLayout_ParseLevel_TestCases))]
+        public void LoggingRule_MinMaxLayout_ParseLevel(string minLevel, string maxLevel, LogLevel[] expectedLevels)
+        {
+            var config = XmlLoggingConfiguration.CreateFromXmlString(@"
+            <nlog>"
+                + (!string.IsNullOrEmpty(minLevel) ? $"<variable name='var_minlevel' value='{minLevel}'/>" : "")
+                + (!string.IsNullOrEmpty(maxLevel) ? $"<variable name='var_maxlevel' value='{maxLevel}'/>" : "") +
+                @"<targets>
+                    <target name='d1' type='Debug' layout='${message}' />
+                </targets>
+                <rules>
+                    <logger name='*' minlevel='${var:var_minlevel}' maxlevel='${var:var_maxlevel}' writeTo='d1' />
+                </rules>
+            </nlog>");
+
+            LogManager.Configuration = config;
+            var logger = LogManager.GetLogger(nameof(LoggingRule_MinMaxLayout_ParseLevel));
+
+            for (int i = LogLevel.MinLevel.Ordinal; i <= LogLevel.MaxLevel.Ordinal; ++i)
+            {
+                if (expectedLevels.Contains(LogLevel.FromOrdinal(i)))
+                    Assert.True(logger.IsEnabled(LogLevel.FromOrdinal(i)));
+                else
+                    Assert.False(logger.IsEnabled(LogLevel.FromOrdinal(i)));
+            }
+
+            // Verify that runtime override also works
+            LogManager.Configuration.Variables["var_minlevel"] = LogLevel.Fatal.ToString();
+            LogManager.Configuration.Variables["var_maxlevel"] = LogLevel.Fatal.ToString();
+            LogManager.ReconfigExistingLoggers();
+
+            for (int i = LogLevel.MinLevel.Ordinal; i <= LogLevel.MaxLevel.Ordinal; ++i)
+            {
+                if (LogLevel.Fatal.Ordinal == i)
+                    Assert.True(logger.IsEnabled(LogLevel.FromOrdinal(i)));
+                else
+                    Assert.False(logger.IsEnabled(LogLevel.FromOrdinal(i)));
+            }
+        }
+
+        public static IEnumerable<object[]> LoggingRule_MinMaxLayout_ParseLevel_TestCases()
```

Comment:
The idea by the [MemberData] was to reduce code duplication and make it easier to maintain. It seems we lost the latter one...

Reply:
You are welcome to unroll all the loops and generate one giant list.

Reply:
Is there a commit which I could use? Otherwise this will take some time.

Reply:
Sorry don't have a commit ready in my pocket :)

Reply:
I see these are now 80 test cases. I think this is hard to maintain and unneeded.  If we introduce a bug, it would be nice if 1 test fails, and not multiple. 

As rule of thumb I try to create these test cases:

- happy path
- alternatives paths 
- corner cases
- cases if needed for coverage.

That means no combination of cases, if it isn't needed for the list above.

For example this test: (shorter test, easier to show)

```c#
      [Theory]
        [InlineData("Off")]
        [InlineData("")]
        [InlineData((string)null)]
        [InlineData("Trace")]
        [InlineData("Debug")]
        [InlineData("Info")]
        [InlineData("Warn")]
        [InlineData("Error")]
        [InlineData(" error")]
        [InlineData("Fatal")]
        [InlineData("Wrong")]
        public void LoggingRule_LevelLayout_ParseLevel(string levelVariable
```
I think the cases "Debug", "Info", "Warn", "Error" (not " error") and "Fatal" not needed (redundant)

Reply:
You are right about not needing all the official Levels. I just wanted to make absolutely sure I didn't have an issue with min/max-levels crossing.